src/main/java/interview/Parser.java

```diff
@@ -0,0 +1,52 @@
+package interview;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+/**
+ * This class is thread safe.
+ */
+public class Parser {
+    private File file;
+
+    public synchronized void setFile(File f) {
```

Comment:
On my opinion it's better to create an object of this class with file passed as arg to its constructor and to make the field `file` final (don't forget to remove setter). In this case the object becomes immutable and no need to sync anything.